src/component/Productdetail.js

```diff
@@ -0,0 +1,240 @@
+  const [selectedId, setSelectedId] = useState(1);
+  const [addedStatus, setAddedStatus] = useState(Boolean);
+  const [qty, setQty] = useState(1);
+  const [{ basket }, dispatch] = useStateValue();
+  const [loading, setLoading] = useState(false);
+
+  const addItemWithStatus = () => {
+    setAddCart(1);
+    addToBasket();
+    Swal.fire({
+      position: "top-end",
+      icon: "success",
+      customClass: "swal-wide",
+      title: selectedProduct.title,
+      text: "Sucessfully Added",
+
+      imageUrl: selectedProduct.image,
+      imageWidth: 40,
+      imageHeight: 40,
+      showConfirmButton: false,
+      showCloseButton: true,
+
+      timer: 2500,
+    });
+  };
+  const addToBasket = async () => {
+    await setAddCart(selectedProduct.id);
+    dispatch({
+      type: "ADD_TO_BASKET",
+      item: {
+        id: selectedProduct.id,
+        title: selectedProduct.title,
+        price: selectedProduct.price,
+        category: selectedProduct.category,
+        description: selectedProduct.description,
+        image: selectedProduct.image,
+        qty: qty,
+      },
+    });
+  };
+  const handleChangeQtyAdd = () => {
+    setQty(qty + 1);
+  };
+  const handleChangeQtySub = () => {
+    setQty(qty - 1);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    const getProducts = async () => {
+      const response = await fetch(
+        `https://fakestoreapi.com/products/${state}`
+      );
+
+      const jsonResponse = await response.json();
+
+      setSelectedProduct(jsonResponse);
+      setSelectedProductImage(jsonResponse.image);
+    };
+
+    getProducts();
+    setLoading(false);
+  }, []);
+
+  const showImage = () => {
+    setSelectedProductImage(selectedProduct.image);
+    setSelectedId(1);
+  };
+
+  const status = basket?.some((o) => o.id === selectedProduct.id);
+
+  return (
+    <div className="productdetalis">
+      <div className="productdetalis__left">
+        <div className="productdetalis__leftList">
+          <img
+            className="productdetalis__leftImage"
+            src={selectedProduct.image}
+            alt=""
+          />
+        </div>
+        <div className="productdetalis__leftdivImagePriview">
+          <div
+            className="productdetalis__leftDivImagePriviewDiv"
+            style={
+              selectedId == 1
+                ? {
+                    border: "2px solid",
+                    borderColor: "#3f2be4",
+                    borderRadius: "20px",
+                  }
+                : {}
+            }
+          >
+            <img
+              onClick={showImage}
+              className="productdetalis__leftdivImagePriviewImage"
+              src={selectedProduct.image}
+              alt=""
+            />
+          </div>
+          <div
+            className="productdetalis__leftDivImagePriviewDiv"
+            style={
+              selectedId == 2
+                ? {
+                    border: "2px solid",
```

Comment:
these styles can be imported via a css class. The class can be added conditionally.
https://stackoverflow.com/questions/39281032/how-to-conditionally-apply-css-classes-in-react-js